applications/ConstitutiveLawsApplication/custom_constitutive_thermal/thermal_elastic_isotropic_3d.cpp

```diff
@@ -0,0 +1,151 @@
+// KRATOS ___                _   _ _         _   _             __                       _
+//       / __\___  _ __  ___| |_(_) |_ _   _| |_(_)_   _____  / /  __ ___      _____   /_\  _ __  _ __
+//      / /  / _ \| '_ \/ __| __| | __| | | | __| \ \ / / _ \/ /  / _` \ \ /\ / / __| //_\\| '_ \| '_  |
+//     / /__| (_) | | | \__ \ |_| | |_| |_| | |_| |\ V /  __/ /__| (_| |\ V  V /\__ \/  _  \ |_) | |_) |
+//     \____/\___/|_| |_|___/\__|_|\__|\__,_|\__|_| \_/ \___\____/\__,_| \_/\_/ |___/\_/ \_/ .__/| .__/
+//                                                                                         |_|   |_|
+//
+//  License:         BSD License
+//                   license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Alejandro Cornejo
+//
+
+// System includes
+#include <iostream>
+
+// External includes
+
+// Project includes
+#include "custom_constitutive_thermal/thermal_elastic_isotropic_3d.h"
+
+#include "includes/checks.h"
+#include "structural_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ThermalElasticIsotropic3D::CalculateMaterialResponsePK2(ConstitutiveLaw::Parameters& rValues)
+{
+    KRATOS_TRY
+    const auto& r_process_info = rValues.GetProcessInfo();
+    if (r_process_info[STEP] == 1) // We storage the ref temperature as the initial one
```

Comment:
I actually do not like this way of taking the reference temperature. 
What if we restart and from a different ref temperature ?
Can't we make a new variable or use an existing variable to set this reference temperature ?

Reply:
Please correct me if I am wrong, this function is called for every element no ? 
So this check if STEP==1 is done for each element right ? This looks expensive !

Reply:
Definitely, as I said at the beginning this is a work in progress PR :D 
In this case I adopted that the ref Temp was the one at the very beginning of the calculation, would you prefer to set it via material property?

Reply:
This is not solved